dev-packages/request-service/src/common-request-service.ts

```diff
@@ -0,0 +1,120 @@
+/********************************************************************************
+ * Copyright (C) 2022 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { CancellationToken } from 'vscode-languageserver-protocol';
```

Comment:
I'm not sure we should use `CancellationToken` from `vscode-languageserver-protocol` here... I know we can't depend on `@theia/core` either.

Can we provide a minimal implementation with this package?

I was looking at [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) but it seems to only be supported on Node starting with 15.0.0.

Reply:
Makes sense 👍 I copied the `CancellationToken` interface, so that users of Theia can use the `CancellationToken` implementation from the core package.